DMCompiler/DMStandard/Types/Particles.dm

```diff
@@ -3,29 +3,36 @@
+	var/transform as opendream_unimplemented // matrix or list. list can be simple matrix, complex matrix or projection matrix. thus: list(a, b, c, d, e, f) OR list(xx,xy,xz, yx,yy,yz, zx,zy,zz) OR list(xx,xy,xz, yx,yy,yz, zx,zy,zz, cx,cy,cz) OR list(xx,xy,xz,xw, yx,yy,yz,yw, zx,zy,zz,zw, wx,wy,wz,ww)
+
 	//Vars that apply when a particle spawns
-	var/lifespan as opendream_unimplemented
-	var/fade as opendream_unimplemented
-	var/fadein as opendream_unimplemented
-	var/icon as opendream_unimplemented // not typed as an /icon because this can also be set to a list of icons (?????)
-	var/icon_state as opendream_unimplemented // can also be a list
-	var/color as opendream_unimplemented
-	var/color_change as opendream_unimplemented
-	var/position as opendream_unimplemented
-	var/velocity as opendream_unimplemented
-	var/scale as opendream_unimplemented
-	var/grow as opendream_unimplemented // The DM ref says this is a num var, yet says its default value is list(0,0), when actually it is null.
-	var/rotation as opendream_unimplemented
-	var/spin as opendream_unimplemented
-	var/friction as opendream_unimplemented
+	var/lifespan as opendream_unimplemented  // actual time a particle exists is fadein + lifespan + fade. thus this just the time it spends fully faded in. null is treated as 
+	var/fade as opendream_unimplemented // null treated as 0
+	var/fadein as opendream_unimplemented // null treated as 0
+	var/icon as opendream_unimplemented // either icon or list(icon = weightofthisicon, icon = weightofthisicon) if null defaults to a 1x1 white pixel
+	var/icon_state as opendream_unimplemented // either string or list(string = weightofthisiconstate, string = weightofthisiconstate) if null defaults to a 1x1 white pixel
+	var/color as opendream_unimplemented // null treated as 0
+	var/color_change as opendream_unimplemented // null treated as 0
+	var/position as opendream_unimplemented // Usually list but if a number treated as list(position, position, position). null is treated as 0
+	var/velocity as opendream_unimplemented // Usually list but if a number treated as list(velocity, velocity, velocity). null is treated as 0
+	var/scale as opendream_unimplemented // if null defaults to 1, if number treated as list(scale, scale)
+	var/grow as opendream_unimplemented // if null defaults to 0, if number treated as list(grow, grow)
+	var/rotation as opendream_unimplemented // null treated as 0
+	var/spin as opendream_unimplemented // null treated as 0
+	var/friction as opendream_unimplemented // null treated as 0, numbers below 0 treated as 0
+
 	//Vars that are evaluated every tick
-	var/list/drift as opendream_unimplemented
+	var/drift as opendream_unimplemented // Usually list but if a number treated as list(drift, drift, drift)
+
+//misc notes
+// particle image height/width is not considered for TILE_BOUND-less atoms
+// no idea how z works, except that by default going under -1 and over 200 makes the particle vanish
```

Comment:
Maybe it has an effect on render order with other particles?

Reply:
I didnt notice that, though I did notice they always layer under ones that spawned before